grakn-engine/src/test/java/ai/grakn/engine/task/postprocessing/RedisIndexStorageTest.java

```diff
@@ -0,0 +1,116 @@
+ * along with Grakn. If not, see <http://www.gnu.org/licenses/gpl.txt>.
+ */
+
+package ai.grakn.engine.task.postprocessing;
+
+import ai.grakn.Keyspace;
+import ai.grakn.concept.ConceptId;
+import ai.grakn.test.rule.InMemoryRedisContext;
+import com.codahale.metrics.MetricRegistry;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.BeforeClass;
+import org.junit.ClassRule;
+import org.junit.Test;
+
+import java.util.Arrays;
+import java.util.HashSet;
+import java.util.Set;
+
+import static org.hamcrest.Matchers.containsInAnyOrder;
+import static org.hamcrest.Matchers.empty;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertThat;
+
+/**
+ * <p>
+ *    Tests that index related post processing data can be stored and retrieved from redis
+ * </p>
+ *
+ * @author fppt
+ */
+public class RedisIndexStorageTest {
+    @ClassRule
+    public static final InMemoryRedisContext IN_MEMORY_REDIS_CONTEXT = InMemoryRedisContext.create();
+
+    private static RedisIndexStorage indexStorage;
+    private final static Keyspace keyspace1 = Keyspace.of("myhappypotato");
+    private final static Keyspace keyspace2 = Keyspace.of("mysadapple");
+    private final static String index1 = "index1";
+    private final static String index2 = "index2";
+    private final static Set<ConceptId> conceptIds = new HashSet<>(Arrays.asList(ConceptId.of("a"), ConceptId.of("b"), ConceptId.of("c")));
+
+    @BeforeClass
+    public static void getConnection(){
+        indexStorage = RedisIndexStorage.create(IN_MEMORY_REDIS_CONTEXT.jedisPool(), new MetricRegistry());
+    }
+
+    @Before
+    public void addDataToStorage(){
+        indexStorage.addIndex(keyspace1, index1, conceptIds);
+        indexStorage.addIndex(keyspace2, index1, conceptIds);
+        indexStorage.addIndex(keyspace2, index2, conceptIds);
+    }
+
+    @After
+    public void removeDataFromStorage(){
+        indexStorage.popIndex(keyspace1);
+        indexStorage.popIndex(keyspace2);
+        indexStorage.popIds(keyspace1, index1);
+        indexStorage.popIds(keyspace2, index1);
+        indexStorage.popIds(keyspace2, index2);
+    }
+
+    @Test
+    public void whenAddingIndices_EnsureKeyspaceListContainIndices(){
+        assertEquals(index1, indexStorage.popIndex(keyspace1));
+        assertEquals(null, indexStorage.popIndex(keyspace1));
+        assertEquals(index1, indexStorage.popIndex(keyspace2));
+        assertEquals(index2, indexStorage.popIndex(keyspace2));
+    }
+
+    @Test
+    public void whenAddingDifferentKeyspaceIndices_EnsureIndicesAreStoredUnderDifferentKeys(){
+        assertJedisContains(RedisIndexStorage.getIndicesKey(keyspace1), index1);
+        assertJedisContains(RedisIndexStorage.getIndicesKey(keyspace2), index1, index2);
+    }
+
+    @Test
+    public void whenPoppingIndex_EnsureListOfIndicesAreUpdated(){
+        assertJedisContains(RedisIndexStorage.getIndicesKey(keyspace2), index1, index2);
```

Comment:
Can you tell me why we need this `assertJedisContains` here?

In my opinion it is a given condition and therefore we should not test it. If let's say, we do want to test the initial state of the indexStorage, we should have a standalone test for it.

Reply:
Hmmm that is a good point. I am just being pedantic to ensure the start of the test state is as we expect it. 

I find this makes it easier to catch certain bugs. For example is someone changes the`@Before` step this would catch it early.